src/Functions/URL/removeGarbageParametersFromURL.cpp

```diff
@@ -0,0 +1,427 @@
+        {
+            x += pow(n[i] - np[i], 2) / np[i];
+        }
+
+        /// Calculating quantile
+        double q;
+        try
+        {
+            boost::math::chi_squared_distribution dist(static_cast<double>(l - 1));
+            q = quantile(dist, 0.95);
+        }
+        catch (std::domain_error &)
+        {
+            return false;
+        }
+
+        /// If Pearson statistics is from 0 to quantile, the distribution is uniform
+        return 0 <= x && x <= q;
+    }
+
+    /// @brief Create a copy of string, move symbols to one interval and check if the distribution is uniform for base64 for URL string
+    /// @param c_start Pointer to the first symbol of the string
+    /// @param c_end Pointer to the symbol following the last symbol of the string
+    /// @return if the parameter should be included in URL
+    static bool checkBase64(const char * c_start, const char * c_end)
+    {
+        char * n_c_start = reinterpret_cast<char *>(malloc(c_end - c_start));
+        const char * n_c_end = n_c_start - c_start + c_end;
+        char * write = n_c_start;
+
+        for (const char * i = c_start; i < c_end; ++i)
+        {
+            char correction = 0;
+            if ('a' <= *i && *i <= 'z')
+            {
+                correction = -6;
+            }
+            else if ('0' <= *i && *i <= '9')
+            {
+                correction = 7;
+            }
+            else if ('-' == *i)
+            {
+                correction = 72;
+            }
+            else if ('_' == *i)
+            {
+                correction = 23;
+            }
+            else if ('=' == *i)
+            {
+                correction = 58;
+            }
+            *write = *i + correction;
+            ++write;
+        }
+
+        bool ret = pearsonIsUniform(n_c_start, n_c_end, '7', 'w');
+        free(n_c_start);
+        return !ret;
+    }
+
+    /// @brief Create a copy of string, move symbols to one interval and check if the distribution is uniform for hex string
+    /// @param c_start Pointer to the first symbol of the string
+    /// @param c_end Pointer to the symbol following the last symbol of the string
+    /// @return if the parameter should be included in URL
+    static bool checkHex(const char * c_start, const char * c_end)
+    {
+        char * n_c_start = reinterpret_cast<char *>(malloc(c_end - c_start));
+        const char * n_c_end = n_c_start - c_start + c_end;
+        char * write = n_c_start;
+
+        for (const char * i = c_start; i < c_end; ++i)
+        {
+            char correction = 0;
+            if ('a' <= *i && *i <= 'f')
+            {
+                correction = -39;
+            }
+            *write = *i + correction;
+            ++write;
+        }
+
+        bool ret = pearsonIsUniform(n_c_start, n_c_end, '0', '?');
+        free(n_c_start);
+        return !ret;
+    }
+
+    /// @brief Check if string is long enough and if it could be hex or base64 for url
```

Comment:
Doxygen-style parameter docs are quite rare in the codebase. I would say if the parameters have expressive enough names then we can remove it and just keep l. 199.